docs/DevelopmentGuide.md

```diff
@@ -65,17 +83,47 @@ Simplest way to run tests is to run `bundle exec rake ci`, which will run the en
 Run the tests for the core library with:
 
 ```
-$ bundle exec rake spec:main
+$ bundle exec rake test:main
 ```
 
 **For integrations**
 
-Integrations which interact with dependencies not listed in the `ddtrace` gemspec will need to load these dependencies to run their tests.
+Integrations which interact with dependencies not listed in the `ddtrace` gemspec will need to load these dependencies to run their tests. Each test task could consist of multiple spec tasks which are executed with different groups of dependencies (likely against different versions or variations).
+
+To get a list of the test tasks, run `bundle exec rake -T test`
+
+To run test, run `bundle exec rake test:<spec_name>`
+
+Take `bundle exec rake test:redis` as example, multiple versions of `redis` from different groups are tested.
+
+```ruby
+TEST_METADATA = {
+  'redis' => {
+    'redis-3' => '✅ 2.1 / ✅ 2.2 / ✅ 2.3 / ✅ 2.4 / ✅ 2.5 / ✅ 2.6 / ✅ 2.7 / ✅ 3.0 / ✅ 3.1 / ✅ 3.2 / ✅ 3.3 / ✅ jruby',
+    'redis-4' => '❌ 2.1 / ❌ 2.2 / ❌ 2.3 / ✅ 2.4 / ✅ 2.5 / ✅ 2.6 / ✅ 2.7 / ✅ 3.0 / ✅ 3.1 / ✅ 3.2 / ✅ 3.3 / ✅ jruby',
+    'redis-5' => '❌ 2.1 / ❌ 2.2 / ❌ 2.3 / ❌ 2.4 / ✅ 2.5 / ✅ 2.6 / ✅ 2.7 / ✅ 3.0 / ✅ 3.1 / ✅ 3.2 / ✅ 3.3 / ✅ jruby'
+  }
+}
+```
+
+**Working with appraisal groups**
+
+Checkout [Apppraisal](https://github.com/thoughtbot/appraisal) to learn the basics.
+
+Groups are defined under `appraisal/` directory and their names are prefixed with Ruby runtime based on the environment. `*.gemfile` and `*.gemfile.lock` from `gemfiles/` directory are generated from those definitions.
 
-To get a list of the spec tasks run `bundle exec rake -T test:`
+To find out existing groups in your environment, run `bundle exec appraisal list`
 
-To run any of the specs above just run `bundle exec rake test:<spec_name>`. Ex: `bundle exec rake test:redis`
+After introducing a new group definition or changing existing one, run `bundle exec appraisal generate` to propagate the changes.
```

Comment:
should we use `bundle exec rake appraisal:generate` and `bundle exec rake appraisal:install` to run it for all the ruby versions?

Reply:
Frankly speaking, I think it would be very frustrating to ask our community contributors to setup all ruby environments, because it take a long time and a lot of storage space to install all of them. 

It would be more encouraging and welcoming to have them setup a specific ruby version that are relevant to them and have it working to get started. The maintainer could handle the rest and ideally automate those mechanical procedure such as appraisal.